content/docs/getting-started/futures.md

```diff
@@ -74,282 +127,105 @@ They cannot because they don't have the actual HTTP response. Instead, they
-# use futures::*;
-# pub struct Doubler<T> {
-#     inner: T,
-# }
-#
-# impl<T> Future for Doubler<T>
-# where T: Future<Item = usize>
-# {
-#     type Item = usize;
-#     type Error = T::Error;
-#
-fn poll(&mut self) -> Result<Async<usize>, T::Error> {
-    let v = try_ready!(self.inner.poll());
-    Ok(Async::Ready(v * 2))
-}
-# }
-# pub fn main() {}
-```
-
-# Returning `NotReady`
-
-The last section handwaved a bit and said that once a Future transitioned to the
-ready state, the executor is notified. This enables the executor to be efficient
-in scheduling tasks.
-
-When a function returns Async::NotReady, it signals that it is currently not in
-a ready state and is unable to complete the operation. It is critical that the
-executor is notified when the state transitions to "ready". Otherwise, the task
-will hang infinitely, never getting run again.
-
-For most future implementations, this is done transitively. When a future
-implementation is a combination of sub futures, the outer future only returns
-`NotReady` when at least one inner future returned `NotReady`. Thus, the outer
-future will transition to a ready state once the inner future transitions to a
-ready state. In this case, the `NotReady` contract is already satisfied as the
-inner future will notify the executor when it becomes ready.
-
-Innermost futures, sometimes called "resources", are the ones responsible for
-notifying the executor. This is done by calling [`notify`] on the task returned
-by [`task::current()`].
-
-We will be exploring implementing resources and the task system in more depth in
-a later section. The key take away here is **do not return `NotReady` unless you
-got `NotReady` from an inner future**.
-
-# A More Complicated Future
-
-Let's look at a slightly more complicated future implementation. In this case, we
-will implement a future that takes a host name, does DNS resolution, then
-establishes a connection to the remote host. We assume a `resolve` function
-exists that looks like this:
+For now it's just important to know that futures have two associated types: `Item`
+and `Error`. `Item` is the type of the value that the `Future` will yield when it
+completes. `Error` is the type of Error that the `Future` may yield if there's an
+error before that causes the `Future` from being able to complete.
+
+Finally, `Future`s have one method named `poll`. We won't go into too much detail
+about `poll` in this section since you don't need to know about `poll` to use futures
+with combinators. The only thing to be aware for now is that `poll` is
+what the runtime will call in order to see if the `Future` is complete yet or not. If
+you're curious: `Async` is an enum with values `Ready(Item)` or `NotReady` which
+informs the runtime of if the future is complete or not.
+
+In a future section, we'll be implementing a `Future` from scratch including writing
+a poll function that properly informs the runtime when the future is complete.
+
+## Streams
```

Comment:
I think this section could use a bit on stream combinators to show how they differ from futures. I would show `map` (because `Future::map` was mentioned earlier), but also `for_each` which returns a future.